src/lua/httpc.lua

```diff
@@ -317,13 +380,26 @@ curl_mt = {
             if not method or not url then
                 error('request(method, url[, body, [options]])')
             end
+
+            opts = opts or {}
+            opts.headers = opts.headers or {}
+            method = method:upper()
+            if method == 'PATCH' or
+               method == 'POST' or
+               method == 'PUT' then
+                local content_type = opts.headers['content-type']
+                content_type = header_value_wo_parameters(content_type)
+                body, content_type = encode_body(body, content_type)
+                opts.headers['content-type'] = content_type
```

Comment:
Will not `application/json; charset=utf-8` stripped to `application/json` by this assignment?